<h1 style = "font-size:3rem; color:blue;"> Tax Collections Update for Fiscal 2023

<h2 style = "font-size:3rem; color:black;"> Collection Month: September 2022 (Month #3)

# 1. Background

This notebook is used to analyze and summarize tax revenues collected for the current fiscal year. 

Section 4 provides updates on collections for the latest month for which we have data, while Section 5 provides updates year-to-date collections. The two sections measure the performance of current tax collections compared to OMB's projections and collections from same period of the prior year.

All dollar values are in millions.

# 2. Load Libraries and Connect to Database

Let's load the libraries needed to establish a connection to and analyze the database.

In [2]:
%load_ext sql
from sqlalchemy import create_engine
import psycopg2

After loading the libraries, let's connect to the databse.

In [3]:
%sql postgresql://postgres:KW88193m@localhost:5432/tax_collection
engine = create_engine('postgresql://postgres:KW88193m@localhost/tax_collection')

# 3. Examine Tables in the Database

Before the actual analysis, run and check the queries in this section to make sure the `collection` and `plan` tables have been updated.

**3a. Verify how many months of collections we have so far in the current fiscal year.**

In [4]:
%%sql

SELECT fiscal_year, c.month_id, collection_month 
FROM collection AS c
LEFT JOIN month_index AS mi
ON c.month_id = mi.month_id
WHERE fiscal_year = (SELECT MAX(fiscal_year) FROM collection)
GROUP BY 2,1,3
ORDER BY 2;

 * postgresql://postgres:***@localhost:5432/tax_collection
3 rows affected.


[(2023, 1, 'July-1'), (2023, 2, 'Aug-1'), (2023, 3, 'Sept')]

**3b. Check all four tables in the database to make sure they contain the data needed.**

### Table 1: Collections

The analysis in this notebook only uses data from the two most recent fiscal years. Check to make sure we have data for all collection months from those two fiscal years.

In [5]:
%%sql

SELECT fiscal_year, month_id, 
       SUM(amount) AS total_taxes
FROM collection
WHERE fiscal_year = (SELECT MAX(fiscal_year) FROM collection) OR
      fiscal_year = (SELECT MAX(fiscal_year)-1 FROM collection)
GROUP BY 1,2
ORDER BY 1 DESC, 2 DESC;

 * postgresql://postgres:***@localhost:5432/tax_collection
18 rows affected.


[(2023, 3, Decimal('5965.305')),
 (2023, 2, Decimal('2205.316')),
 (2023, 1, Decimal('16079.560')),
 (2022, 15, Decimal('40.651')),
 (2022, 14, Decimal('190.835')),
 (2022, 13, Decimal('268.608')),
 (2022, 12, Decimal('4813.073')),
 (2022, 11, Decimal('2037.441')),
 (2022, 10, Decimal('6728.369')),
 (2022, 9, Decimal('5687.109')),
 (2022, 8, Decimal('2730.461')),
 (2022, 7, Decimal('7979.459')),
 (2022, 6, Decimal('11758.263')),
 (2022, 5, Decimal('2094.527')),
 (2022, 4, Decimal('2965.665')),
 (2022, 3, Decimal('5654.174')),
 (2022, 2, Decimal('1829.584')),
 (2022, 1, Decimal('14966.097'))]

### Table 2: Plan

Check to make sure the plan table has been updated for the current fiscal year.

In [6]:
%%sql

SELECT fiscal_year, month_id, 
       SUM(amount) AS total_taxes
FROM plan
WHERE fiscal_year = (SELECT MAX(fiscal_year) FROM plan)
GROUP BY 1,2
ORDER BY 1 DESC, 2 DESC;

 * postgresql://postgres:***@localhost:5432/tax_collection
3 rows affected.


[(2023, 3, Decimal('5750.147')),
 (2023, 2, Decimal('1878.820')),
 (2023, 1, Decimal('15789.955'))]

### Table 3: Month Index

This table contains details about the `month_id` field in the `collection` table. 

In [25]:
%%sql

SELECT * FROM month_index

 * postgresql://postgres:***@localhost:5432/tax_collection
15 rows affected.


[(1, 'July-1', 'July'),
 (2, 'Aug-1', 'August'),
 (3, 'Sept', 'September'),
 (4, 'Oct', 'October'),
 (5, 'Nov', 'November'),
 (6, 'Dec', 'December'),
 (7, 'Jan', 'January'),
 (8, 'Feb', 'February'),
 (9, 'Mar', 'March'),
 (10, 'Apr', 'April'),
 (11, 'May', 'May'),
 (12, 'Jun', 'June'),
 (13, 'Jul-2', 'July'),
 (14, 'Aug-2', 'August'),
 (15, 'Adjustment', 'September')]

### Table 4: Tax Index

This table contains detail grouping of individual taxes. 

In [17]:
%%sql

SELECT * FROM tax_index
LIMIT 10

 * postgresql://postgres:***@localhost:5432/tax_collection
10 rows affected.


[('Sales', 'Sales', 'Sales'),
 ('Hotel', 'Hotel', 'Hotel'),
 ('Utility', 'Utility', 'Utility'),
 ('PIT Withholding', 'PIT', 'PIT'),
 ('PIT Estimated', 'PIT', 'PIT'),
 ('PIT Refunds', 'PIT', 'PIT'),
 ('PIT Total (net of refunds)', 'PIT', 'PIT'),
 ('RPT', 'Real Property', 'Real Property'),
 ('MRT', 'Transaction', 'Transaction'),
 ('RPTT', 'Transaction', 'Transaction')]

# 4. Analyze Collections for the Latest Month

Let us analyze collections for the latest month we have data for. I focus on the three things below:
    
    1. Total tax collections for the month compared to plan for the month.
    
    2. Total tax collections for the month compared to same month last year.
    
    3. Breakdown collections compared to plan for individual taxes.
    
    4. Breakdown collections compared to last year's for individual taxes.

## 4.1. Total Taxes: Month Actual vs Month Plan

This query calculates and compares total actual collections to projections for the month. It also calculatesthe two aforementioned numbers,   as well as the dollar and percent difference between the two.

In [5]:
%%sql

WITH latest_collection AS (
SELECT fiscal_year, lm.month_id, mi.collection_month, month_actual 
FROM 
     (SELECT fiscal_year, month_id, SUM(amount) AS month_actual 
      FROM collection
      WHERE fiscal_year = (SELECT MAX(fiscal_year) FROM collection)
      GROUP BY fiscal_year, month_id
      ORDER BY month_id DESC
      LIMIT 1) AS lm
INNER JOIN month_index AS mi
ON lm.month_id = mi.month_id)


SELECT *, 
      (month_actual - month_plan) AS actual_vs_plan,
       ROUND(((month_actual/month_plan)-1)*100, 1) AS pct_diff_plan
FROM
    (SELECT lc.fiscal_year, lc.month_id, lc.collection_month, 
            month_actual, SUM(amount) AS month_plan
     FROM latest_collection AS lc
     LEFT JOIN plan AS pl
     ON lc.fiscal_year = pl.fiscal_year
     AND lc.month_id = pl.month_id
     GROUP BY 1,2,3,4) sub

 * postgresql://postgres:***@localhost:5432/tax_collection
1 rows affected.


[(2023, 3, 'Sept', Decimal('5965.305'), Decimal('5750.147'), Decimal('215.158'), Decimal('3.7'))]

Total collections for the third collection month (September) of Fiscal 2023 was approximately \\$6 billion, which was over \\$215 million (or 3.7 percent) more than projected in the November Plan.

## 4.2. Total Taxes: Month Actual vs Last Year (Same Month)

This query calculates and compares total collections for the month to total collections from the same month last year.

In [6]:
%%sql

WITH ly_comp AS (
SELECT fiscal_year, lm.month_id, 
       mi.collection_month, month_actual,
       month_actual - FIRST_VALUE(month_actual) OVER(ORDER BY fiscal_year) AS diff_ly,
       ROUND((month_actual / FIRST_VALUE(month_actual) 
              OVER(ORDER BY fiscal_year) - 1)*100, 2) AS pchg_ly
FROM 
    (SELECT fiscal_year, month_id, SUM(amount) AS month_actual 
      FROM collection
      WHERE fiscal_year = (SELECT MAX(fiscal_year) FROM collection) AND
            month_id = (SELECT MAX(month_id) FROM collection
                        WHERE fiscal_year = (SELECT MAX(fiscal_year) FROM collection)) 
            OR
            fiscal_year = (SELECT MAX(fiscal_year)-1 FROM collection) AND
            month_id = (SELECT MAX(month_id) FROM collection
                        WHERE fiscal_year = (SELECT MAX(fiscal_year) FROM collection))
      GROUP BY 1,2) AS lm   
INNER JOIN month_index AS mi
ON lm.month_id = mi.month_id)

SELECT * FROM ly_comp
WHERE fiscal_year = '2023';

 * postgresql://postgres:***@localhost:5432/tax_collection
1 rows affected.


[(2023, 3, 'Sept', Decimal('5965.305'), Decimal('311.131'), Decimal('5.50'))]

Total collections for September of Fiscal 2023 (third month) was \\$311 million (or 5.5 percent) more than September of Fiscal 2022.

## 4.3. Individual Taxes (Grouped): Month Actuals vs Month Plan

This query calculates and compares actual collections for each taxes to projections for that tax. It breaks down Section 4.1 by individual tax components.

In [7]:
%%sql

WITH ind_tax AS (
SELECT s2.fiscal_year, s2.month_id, s2.collection_month, s2.tax, 
       month_actual, SUM(amount) AS month_plan
FROM
    (SELECT fiscal_year, s1.month_id, mi.collection_month, tax, month_actual 
     FROM 
         (SELECT fiscal_year, month_id, tax, SUM(amount) AS month_actual  
          FROM collection
          WHERE fiscal_year = (SELECT MAX(fiscal_year) FROM collection) AND 
                month_id = (SELECT MAX(month_id) FROM collection
                            WHERE fiscal_year = (SELECT MAX(fiscal_year) FROM collection))
          GROUP BY 1,2,3
          ORDER BY 3 DESC) AS s1
     INNER JOIN month_index AS mi
     ON s1.month_id = mi.month_id
     ORDER BY 4) AS s2 
    LEFT JOIN plan AS pl
ON s2.fiscal_year = pl.fiscal_year
AND s2.month_id = pl.month_id
AND s2.tax = pl.tax
GROUP BY 1,2,3,4,5)

SELECT *, (month_actual - month_plan) actual_vs_plan,
       ROUND(((month_actual/month_plan)-1)*100, 1) pct_diff_plan
FROM
        (SELECT it.fiscal_year, it.month_id, 
                it.collection_month, tax_category1, 
                SUM(month_actual) month_actual, SUM(month_plan) month_plan
        FROM ind_tax AS it
        LEFT JOIN tax_index AS ti
        ON it.tax = ti.tax
        GROUP BY 1,2,3,4) sub
ORDER BY 8 DESC;

 * postgresql://postgres:***@localhost:5432/tax_collection
12 rows affected.


[(2023, 3, 'Sept', 'Hotel', Decimal('149.534'), Decimal('110.089'), Decimal('39.445'), Decimal('35.8')),
 (2023, 3, 'Sept', 'Business Corporation', Decimal('1172.220'), Decimal('910.314'), Decimal('261.906'), Decimal('28.8')),
 (2023, 3, 'Sept', 'Utility', Decimal('37.480'), Decimal('30.940'), Decimal('6.540'), Decimal('21.1')),
 (2023, 3, 'Sept', 'Audits', Decimal('94.044'), Decimal('78.871'), Decimal('15.173'), Decimal('19.2')),
 (2023, 3, 'Sept', 'UBT', Decimal('472.168'), Decimal('397.995'), Decimal('74.173'), Decimal('18.6')),
 (2023, 3, 'Sept', 'Commercial Rent', Decimal('209.000'), Decimal('191.241'), Decimal('17.759'), Decimal('9.3')),
 (2023, 3, 'Sept', 'Transaction', Decimal('232.000'), Decimal('224.916'), Decimal('7.084'), Decimal('3.1')),
 (2023, 3, 'Sept', 'Sales', Decimal('858.454'), Decimal('874.563'), Decimal('-16.109'), Decimal('-1.8')),
 (2023, 3, 'Sept', 'Others', Decimal('28.091'), Decimal('29.203'), Decimal('-1.112'), Decimal('-3.8')),
 (2023, 3, 'Sept', 'Real Property', Decimal('1456.000'), Decimal('1530.022'), Decimal('-74.022'), Decimal('-4.8')),
 (2023, 3, 'Sept', 'PIT', Decimal('1254.816'), Decimal('1370.358'), Decimal('-115.542'), Decimal('-8.4')),
 (2023, 3, 'Sept', 'Cigarette', Decimal('1.498'), Decimal('1.635'), Decimal('-0.137'), Decimal('-8.4'))]

## 4.4. Individual Taxes (Grouped): Monthly Actuals vs Last Year

This query calculates and compares actual collections for individual taxes to collections from same time last year. It breaks down Section 4.2 by individual tax components.

In [64]:
%%sql

WITH ind_tax AS (
SELECT s2.fiscal_year, s2.month_id, s2.collection_month, tax_category1, 
       SUM(month_actual) month_actual
FROM
    (SELECT fiscal_year, s1.month_id, mi.collection_month, tax, month_actual 
     FROM 
        (SELECT fiscal_year, month_id, tax, SUM(amount) AS month_actual  
         FROM collection
         WHERE fiscal_year = (SELECT MAX(fiscal_year) FROM collection) AND 
               month_id = (SELECT MAX(month_id) FROM collection
                           WHERE fiscal_year = (SELECT MAX(fiscal_year) FROM collection)) OR
                                 fiscal_year = (SELECT MAX(fiscal_year)-1 FROM collection) 
         AND
               month_id = (SELECT MAX(month_id) FROM collection
                           WHERE fiscal_year = (SELECT MAX(fiscal_year) FROM collection))
         GROUP BY 1,2,3
         ORDER BY 3 DESC) s1
         INNER JOIN month_index AS mi
         ON s1.month_id = mi.month_id
         ORDER BY 4) s2
LEFT JOIN tax_index AS ti
ON s2.tax = ti.tax
GROUP BY 4,1,2,3)

SELECT * FROM
    (SELECT *,
        month_actual - FIRST_VALUE(month_actual) 
                        OVER(PARTITION BY tax_category1 ORDER BY fiscal_year ) AS diff_ly,
        ROUND((month_actual / FIRST_VALUE(NULLIF(month_actual, 0))
            OVER(PARTITION BY tax_category1 ORDER BY fiscal_year) - 1)*100, 2) AS pchg_ly
    FROM ind_tax
    GROUP BY 4,1,2,3,5) AS s3
WHERE fiscal_year = (SELECT MAX(fiscal_year) FROM ind_tax)
ORDER BY 7 DESC;

 * postgresql://postgres:***@localhost:5432/tax_collection
12 rows affected.


[(2023, 3, 'Sept', 'Hotel', Decimal('149.534'), Decimal('137.996'), Decimal('1196.01')),
 (2023, 3, 'Sept', 'Audits', Decimal('94.044'), Decimal('51.537'), Decimal('121.24')),
 (2023, 3, 'Sept', 'Utility', Decimal('37.480'), Decimal('7.142'), Decimal('23.54')),
 (2023, 3, 'Sept', 'UBT', Decimal('472.168'), Decimal('51.025'), Decimal('12.12')),
 (2023, 3, 'Sept', 'Sales', Decimal('858.454'), Decimal('89.052'), Decimal('11.57')),
 (2023, 3, 'Sept', 'Transaction', Decimal('232.000'), Decimal('15.000'), Decimal('6.91')),
 (2023, 3, 'Sept', 'Real Property', Decimal('1456.000'), Decimal('77.000'), Decimal('5.58')),
 (2023, 3, 'Sept', 'Commercial Rent', Decimal('209.000'), Decimal('10.000'), Decimal('5.03')),
 (2023, 3, 'Sept', 'Business Corporation', Decimal('1172.220'), Decimal('-16.986'), Decimal('-1.43')),
 (2023, 3, 'Sept', 'PIT', Decimal('1254.816'), Decimal('-106.064'), Decimal('-7.79')),
 (2023, 3, 'Sept', 'Cigarette', Decimal('1.498'), Decimal('-0.219'), Decimal('-12.75')),
 (2023, 3, 'Sept', 'Others', Decimal('28.091'), Decimal('-4.352'), Decimal('-13.41'))]

# 5. Year-to-Date Tax Collection

## 5.1. Total Taxes: YTD Actual vs YTD Plan

This query compares total YTD collections to what was projected.

In [39]:
%%sql

WITH ytd AS(
SELECT s1.fiscal_year, s1.ytd_actual, SUM(amount) AS ytd_plan
FROM
    (SELECT fiscal_year, SUM(amount) AS ytd_actual 
    FROM collection
    WHERE fiscal_year = (SELECT MAX(fiscal_year) FROM collection)
    GROUP BY 1) s1
LEFT JOIN plan AS pl
ON s1.fiscal_year = pl.fiscal_year
GROUP BY 1,2)


SELECT *, (ytd_actual - ytd_plan) AS diff_plan,
       ROUND(((ytd_actual/ytd_plan)-1)*100, 1) AS pchg_plan
FROM ytd
GROUP BY 1,2,3

 * postgresql://postgres:***@localhost:5432/tax_collection
1 rows affected.


[(2023, Decimal('24250.181'), Decimal('23418.922'), Decimal('831.259'), Decimal('3.5'))]

Year-to-date tax collections for Fiscal 2023 have outperformed plan by \\$831 million (or 3.5 percent).

## 5.2. Total Taxes: YTD Actuals vs Last Year (same period)

This query compares YTD total collections to YTD collections from same period last year.

In [65]:
%%sql

WITH ly_comp AS (
SELECT fiscal_year, SUM(amount) AS ytd_actual 
FROM collection
WHERE fiscal_year = (SELECT MAX(fiscal_year) FROM collection) AND
      month_id <= (SELECT MAX(month_id) FROM collection
                   WHERE fiscal_year = (SELECT MAX(fiscal_year) FROM collection)) 
      OR
      fiscal_year = (SELECT MAX(fiscal_year)-1 FROM collection) AND
      month_id <= (SELECT MAX(month_id) FROM collection
                   WHERE fiscal_year = (SELECT MAX(fiscal_year) FROM collection))
GROUP BY 1)

SELECT *
FROM
    (SELECT *, 
            ytd_actual - FIRST_VALUE(ytd_actual) OVER(ORDER BY fiscal_year) AS diff_ly,
            ROUND((ytd_actual / FIRST_VALUE(ytd_actual) 
               OVER(ORDER BY fiscal_year) - 1)*100, 1) AS pchg_ly
    FROM ly_comp AS lc
    GROUP BY 1,2) s1
WHERE fiscal_year = (SELECT MAX(fiscal_year) FROM ly_comp);

 * postgresql://postgres:***@localhost:5432/tax_collection
1 rows affected.


[(2023, Decimal('24250.181'), Decimal('1800.326'), Decimal('8.0'))]

Year-to-date tax collections are up \\$1.8 billion (or eight percent) compared to same period last year.

## 5.3. Individual Taxes (Grouped): YTD Actual vs YTD Plan

This query calculates and compares YTD collections for individual taxes to projections.

In [47]:
%%sql

WITH ytd AS(
SELECT s2.fiscal_year, tax_category1, 
       SUM(ytd_actual) ytd_actual, 
       SUM(ytd_plan) ytd_plan
FROM    
    (SELECT s1.fiscal_year, s1.tax, 
            s1.ytd_actual, SUM(pl.amount) AS ytd_plan
     FROM    
        (SELECT fiscal_year, tax, SUM(amount) AS ytd_actual 
         FROM collection
         WHERE fiscal_year = (SELECT MAX(fiscal_year) FROM collection)
         GROUP BY 1,2) s1    
    LEFT JOIN plan AS pl
    ON s1.tax = pl.tax 
    AND s1.fiscal_year = pl.fiscal_year
    GROUP BY 1,2,3) s2   
JOIN tax_index AS ti
ON s2.tax = ti.tax
GROUP BY 1,2
          )

SELECT *, 
       (ytd_actual - ytd_plan) AS diff_plan,
       ROUND(((ytd_actual/ytd_plan)-1)*100, 1) AS pchg_plan
FROM ytd
GROUP BY 1,2,3,4
ORDER BY 6 DESC;

 * postgresql://postgres:***@localhost:5432/tax_collection
12 rows affected.


[(2023, 'Others', Decimal('218.760'), Decimal('129.777'), Decimal('88.983'), Decimal('68.6')),
 (2023, 'Hotel', Decimal('149.534'), Decimal('110.089'), Decimal('39.445'), Decimal('35.8')),
 (2023, 'Business Corporation', Decimal('1172.220'), Decimal('910.314'), Decimal('261.906'), Decimal('28.8')),
 (2023, 'Audits', Decimal('94.044'), Decimal('78.871'), Decimal('15.173'), Decimal('19.2')),
 (2023, 'UBT', Decimal('472.168'), Decimal('397.995'), Decimal('74.173'), Decimal('18.6')),
 (2023, 'Transaction', Decimal('779.159'), Decimal('674.748'), Decimal('104.411'), Decimal('15.5')),
 (2023, 'Utility', Decimal('72.865'), Decimal('64.847'), Decimal('8.018'), Decimal('12.4')),
 (2023, 'Sales', Decimal('2252.742'), Decimal('2039.726'), Decimal('213.016'), Decimal('10.4')),
 (2023, 'Commercial Rent', Decimal('209.000'), Decimal('191.241'), Decimal('17.759'), Decimal('9.3')),
 (2023, 'PIT', Decimal('2939.740'), Decimal('2935.174'), Decimal('4.566'), Decimal('0.2')),
 (2023, 'Real Property', Decimal('15885.194'), Decimal('15881.107'), Decimal('4.087'), Decimal('0.0')),
 (2023, 'Cigarette', Decimal('4.755'), Decimal('5.033'), Decimal('-0.278'), Decimal('-5.5'))]

## 5.4. Individual Taxes (Grouped): YTD Actual vs YTD Plan

This query calculates and compares YTD collections for the individual taxes to YTD collections from same period last year.

In [49]:
%%sql

WITH ly_comp AS (
SELECT s1.fiscal_year, tax_category1, SUM(s1.ytd_actual) AS ytd_actual
FROM
   (SELECT fiscal_year, tax, SUM(amount) AS ytd_actual 
    FROM collection
    WHERE fiscal_year = (SELECT MAX(fiscal_year) FROM collection) AND
          month_id <= (SELECT MAX(month_id) FROM collection
                       WHERE fiscal_year = (SELECT MAX(fiscal_year) FROM collection)) 
          OR
          fiscal_year = (SELECT MAX(fiscal_year)-1 FROM collection) AND
          month_id <= (SELECT MAX(month_id) FROM collection
                       WHERE fiscal_year = (SELECT MAX(fiscal_year) FROM collection))
    GROUP BY 1,2
    ORDER BY 2,1) s1
JOIN tax_index AS ti
ON s1.tax = ti.tax
GROUP BY 2,1
                )


SELECT *
FROM
    (SELECT *, 
            ytd_actual - FIRST_VALUE(ytd_actual) OVER(PARTITION BY tax_category1 
                                                 ORDER BY fiscal_year) AS diff_ly,
            ROUND((ytd_actual / FIRST_VALUE(ytd_actual) OVER(PARTITION BY tax_category1 
                                                        ORDER BY fiscal_year) - 1)*100, 1
                 ) AS pchg_ly
     FROM ly_comp AS lc
     GROUP BY 1,2,3) s1
WHERE fiscal_year = (SELECT MAX(fiscal_year) FROM ly_comp)
ORDER BY pchg_ly DESC;

 * postgresql://postgres:***@localhost:5432/tax_collection
12 rows affected.


[(2023, 'Hotel', Decimal('149.534'), Decimal('137.996'), Decimal('1196.0')),
 (2023, 'Audits', Decimal('94.044'), Decimal('51.537'), Decimal('121.2')),
 (2023, 'Others', Decimal('218.760'), Decimal('86.988'), Decimal('66.0')),
 (2023, 'Sales', Decimal('2252.742'), Decimal('341.019'), Decimal('17.8')),
 (2023, 'Utility', Decimal('72.865'), Decimal('9.357'), Decimal('14.7')),
 (2023, 'Transaction', Decimal('779.159'), Decimal('95.159'), Decimal('13.9')),
 (2023, 'UBT', Decimal('472.168'), Decimal('51.025'), Decimal('12.1')),
 (2023, 'Real Property', Decimal('15885.194'), Decimal('942.194'), Decimal('6.3')),
 (2023, 'Commercial Rent', Decimal('209.000'), Decimal('10.000'), Decimal('5.0')),
 (2023, 'PIT', Decimal('2939.740'), Decimal('92.640'), Decimal('3.3')),
 (2023, 'Business Corporation', Decimal('1172.220'), Decimal('-16.986'), Decimal('-1.4')),
 (2023, 'Cigarette', Decimal('4.755'), Decimal('-0.603'), Decimal('-11.3'))]